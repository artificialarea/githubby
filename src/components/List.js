import { useState, useEffect } from 'react';
import ApiService from '../services/api-service';
import ListItem from './ListItem';

export default function List(props) {
  const { owner, repo } = props.match.params;
  const [commits, setCommits] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    setError(null);
    setCommits([]);
    ApiService.getRepo(owner, repo)
      .then(res => {
        setCommits(res);
        setError(null);
      })
      .catch(res => {
        setError(res.message);
        // setError('Repo not found ( >_<)');
      })

  }, [owner, repo]);

  return (
    <>
      <h2>{owner} / {repo} {error && <span className='error'>{error}</span>}</h2>
      <ul className="list">
        {commits.map((item) => (
          <ListItem 
            key={item.sha}
            item={item}
          />
        ))}
      </ul>
    </>
  )
}

List.defaultProps = {
  commits: []
} 