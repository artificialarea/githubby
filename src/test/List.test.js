import React from 'react';
import ReactDOM from 'react-dom';
import List from '../components/List';
import { BrowserRouter } from 'react-router-dom';

describe.skip('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <List />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })
})