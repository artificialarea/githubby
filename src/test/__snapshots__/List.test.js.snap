// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders a list 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <List
    commits={
      Array [
        Object {
          "author": Object {
            "avatar_url": "https://avatars.githubusercontent.com/u/1991943?v=4",
            "login": "artificialarea",
          },
          "commit": Object {
            "author": Object {
              "date": "2020-08-17T23:01:44Z",
            },
            "message": "test commit message",
          },
          "sha": "c32035e6e979e0d83ceecda1e3d175d4449d73dd",
        },
        Object {
          "author": Object {
            "avatar_url": "https://avatars.githubusercontent.com/u/1991943?v=4",
            "login": "artificialarea",
          },
          "commit": Object {
            "author": Object {
              "date": "2020-08-18T23:01:44Z",
            },
            "message": "test commit message 2",
          },
          "sha": "c32035e6e979e0d83ceecda1e3d175d4449d73zz",
        },
      ]
    }
  />
</Router>
`;

exports[`<List /> renders something 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <List
    commits={Array []}
  />
</Router>
`;
